extensions/jms/src/main/java/org/mule/extensions/jms/api/message/JmsHeaders.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extensions.jms.api.message;
+
+import org.mule.extensions.jms.api.destination.JmsDestination;
+
+import javax.jms.DeliveryMode;
+
+/**
+ * JMS header fields contain values used by both clients and providers to identify and route messages.
+ * A message's complete header is transmitted to all JMS clients that receive the message.
+ *
+ * @since 4.0
+ */
+public interface JmsHeaders {
+
+  /**
+   * @return the destination to which the message is being sent
+   */
+  JmsDestination getJMSDestination();
+
+  /**
+   * @return the delivery mode specified when the message was sent, which
+   * can be either {@link DeliveryMode#PERSISTENT} or {@link DeliveryMode#NON_PERSISTENT}
+   */
+  int getJMSDeliveryMode();
+
+  /**
+   * JMS provider calculates its expiration time by adding the {@code timeToLive}
+   * value specified on the send method to the time the message was sent (for transacted sends,
+   * this is the time the client sends the message, not the time the transaction is committed)
+   * <p>
+   * If the {@code timeToLive} is specified as {@code zero}, the message's expiration time is
+   * set to zero to indicate that the message does not expire.
+   *
+   * @return the message's expiration time or {@code zero} if the message does not expire
+   */
+  long getJMSExpiration();
```

Comment:
Do this headers always have values?. I'm asking about all the ones that return primitive types